File: backend/src/main/java/com/trojanscheduler/project/model/Course.java
package com.trojanscheduler.project.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity(name="courses")
@Data
public class Course {
    @Id
    private int id;
    private String department;
    private String courseNumber;
    private String courseName;
    private String units;

    @Column(columnDefinition = "TEXT")
    private String description;
    private String term;

    @OneToMany(mappedBy = "course")
    private List<Section> sectionList = new ArrayList<>();

    public int hashCode() {
        return Objects.hash(id, department, courseNumber);
    }


}

File: get_course_data/model/course.py
import dataclasses
from typing import List

from model.section import Section

@dataclasses.dataclass
class Course:
    department: str = ""
    course_number: str = ""
    course_name: str = ""
    units: str = ""
    description: str = ""
    term: str = ""
    sections: List[Section] = dataclasses.field(default_factory=list)

    def to_dict(self):
        return {
            "term": self.term,
            "department": self.department,
            "course_number": self.course_number,
            "course_name": self.course_name,
            "units": self.units,
            "description": self.description,
            "sections": [section.to_dict() for section in self.sections]
        }

    def __str__(self):
        return f"{self.term} {self.department}-{self.course_number}: {self.course_name} {self.units}" + \
                f"\n\t{self.description}" + str(self.sections)



